Core/Src/alerts.c

```diff
@@ -0,0 +1,79 @@
+/* ==================================================================== */
+/* ============================= INCLUDES ============================= */
+/* ==================================================================== */
+#include "alerts.h"
+
+
+AlertStatus_E alertStatus(Alert_S* alert)
+{
+    return alert->alertStatus;
+}
+
+void runAlertMonitor(Bms_S* bms, Alert_S* alert)
+{
+    if (alert->alertStatus == ALERT_CLEARED)
+    {
+        // Determine if we need to set the alert
+        if (alert->alertConditionPresent(bms))
+        {
+            // Increment alert timer by 10ms
+            updateTimer_10ms(&alert->alertTimer);
+        }
+        else
+        {
+            // Reset alert timer
+            clearTimer(&alert->alertTimer);
+        }
+
+        // Determine if alert was set
+        if (checkTimerExpired(&alert->alertTimer))
+        {
+            // Timer expired - Set alert
+            alert->alertStatus = ALERT_SET;
+            // Load timer with alert clear time
+            configureTimer(&alert->alertTimer, alert->clearTime_MS);
+        }
+
+    }
+    else if (alert->alertStatus == ALERT_SET)
+    {
+        // Determine if we can clear the alert
+        if (!alert->alertConditionPresent(bms))
+        {
+            // Increment clear timer by 10ms
+            updateTimer_10ms(&alert->alertTimer);
+        }
+        else
+        {
+            // Alert conditions detected. Reset clear timer
+            clearTimer(&alert->alertTimer);
+        }
+
+        // Determine if alert was cleared
+        if (checkTimerExpired(&alert->alertTimer))
+        {
+            // Timer expired - Clear alert
+            alert->alertStatus = ALERT_CLEARED;
+            // Load timer with alert set time
+            configureTimer(&alert->alertTimer, alert->setTime_MS);
+        }
+    }
```

Comment:
Handle else condition where alertStatus is neither set nor cleared. ( should never happen)